ror/spec/requests/tags_request_spec.rb

```diff
@@ -0,0 +1,176 @@
+    end
+
+    context 'with a tag name already existed' do
+      let(:expected) { { 'name' => ['has already been taken'] } }
+
+      before { post '/tags', params: { 'tag[name]' => existed_tag_name } }
+
+      it('returns an error message') { expect(JSON.parse(response.body)['errors']).to include(expected) }
+    end
+  end
+
+  describe 'PUT /tags' do
+    let(:updated_tag_name) { 'updatedTagName' }
+
+    context 'with a valid tag id' do
+      before { put "/tags/#{@existed_tag_id}", params: { 'tag[name]' => updated_tag_name } }
+
+      it('updates the name of the tag') { expect(JSON.parse(response.body)['tag']['name']).to eq(updated_tag_name) }
+    end
+
+    context 'with an invalid tag id' do
+      let(:expected) { {'id' => ['errors.messages.not_found']} }
+
+      before { put "/tags/#{invalid_tag_id}", params: { :name => updated_tag_name } }
+
+      it('returns an error message') { expect(JSON.parse(response.body)['errors']).to eq(expected) }
+    end
+  end
+
+  describe 'GET /tags' do
+    context 'with the filter query param' do
+      let(:wrong_filter_query) { 'fooBar' }
+      let(:right_filter_query) { 'existed' }
+
+      context 'with a filter that matches the tag name' do
+        before { get "/tags?filter=#{right_filter_query}" }
+
+        it('returns tags containing the tag') { expect(JSON.parse(response.body)['tags']).to include(@existed_tag_object) }
+      end
+
+      context 'with a filter that does not match any tag names' do
+        before { get "/tags?filter=#{wrong_filter_query}" }
+
+        it('returns tags not containing the tag') { expect(JSON.parse(response.body)['tags']).not_to include(@existed_tag_object) }
+      end
+    end
+
+    context 'without the filter query param' do
+      before { get '/tags' }
+
+      it('has the length equals to the target length') { expect(JSON.parse(response.body)['tags'].size).to eq(@existed_tags_length) }
+      it('returns tags containing the tag')            { expect(JSON.parse(response.body)['tags']).to include(@existed_tag_object) }
+    end
+  end
+
+  #
+  # Nested resources
+  #
+  let(:existed_article_title)  { 'existedArticleTitle' }
+  let(:existed_article_body)   { 'existedArticleBody' }
+  let(:existed_article_status) { 'public' }
+
+  @existed_article_object = nil
+  @existed_article_id     = nil
+
+  before do
+    post '/articles', params: {
+      "article[title]"  => existed_article_title,
+      "article[body]"   => existed_article_body,
+      "article[status]" => existed_article_status
+    }
+
+    @existed_article_object = JSON.parse(response.body)['article']
+    @existed_article_id     = @existed_article_object['id']
+  end
+
+  describe 'POST /articles/:article_id/tags' do
+    context 'with a valid article id' do
+      @returned_tag = nil
+
+      before do
+        post "/articles/#{@existed_article_id}/tags", params: { :tag_id => @existed_tag_id }
```

Comment:
Stylewise, if you have a hash with symbols as keys, you would use the colon syntax instead of the rocket syntax:

`params: { :tag_id => @existed_tag_id } ` becomes `params: { tag_id: @existed_tag_id }`

Reply:
Fixed.